src/main/controllers/UpdatePreferenceController.ts

```diff
@@ -0,0 +1,30 @@
+import { Request, Response } from 'express';
+
+export default class UpdatePreferenceController {
+
+  public get(req: Request, res: Response): void {
+    res.render('update-preference', {
+      ...(req.t('update-preference', { returnObjects: true })),
+    });
+  }
+
+  public post(req: Request, res: Response): void {
+    if (req.body['update-preference'] === 'email' && req.body['saveButton'] === 'saveContinue') {
+      res.redirect('/would-you-want-to-take-part-in-video-hearings');
+    }
+    else if (req.body['update-preference'] === 'post' && req.body['saveButton'] === 'saveContinue') {
+      res.redirect('/would-you-want-to-take-part-in-video-hearings');
+    }
```

Comment:
Maybe we can merge these together? It also looks like it's something that could be separated into a separate function. Same for the other two `if` statements.

Reply:
Tidied up and replaced if-else with conditional operator.